### PR TITLE
[react-simple-maps] Add missing hook types

### PR DESCRIPTION
**Pull request by piercemorris**
Relates to [this comment](https://github.com/zcreativelabs/react-simple-maps/issues/334#issuecomment-1769467041) where the hooks types are still undefined, resulting in ts(2724) linter error

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by piercemorris** [comment]
Anyone able to take a look at this?

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-simple-maps/react-simple-maps-tests.tsx

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9124cedb6b164e5e9cc666c9eb815b29a12965e9
</p>

File: types/react-simple-maps/index.d.ts [hunk 1]
```diff
@@ -88,6 +89,38 @@ export interface ZoomableGroupProps extends React.SVGAttributes<SVGGElement> {
     translateExtent?: [[number, number], [number, number]] | undefined;
 }
 
+interface useZoomPanProps {
```
Comment (jakebailey) [inline types/react-simple-maps/index.d.ts:92]: Is this interface intentionally lowercased? Typically these are uppercase (note the previous examples).

File: types/react-simple-maps/package.json [hunk 2]
```diff
@@ -26,6 +26,10 @@
         {
             "name": "Kouame Komenan ",
             "githubUsername": "komenank"
+        },
+        {
+            "name": "Pierce Morris",
+            "githubUsername": "piercejmorris"
```
Comment (jakebailey) [inline types/react-simple-maps/package.json:32]: Are you intending on being an owner of this package, doing reviews, etc? This is more about DT infra than it is credit.

File: types/react-simple-maps/index.d.ts (hunk 1 above)
Comment (piercemorris) [reply types/react-simple-maps/index.d.ts:92]: No not intentional, I will update it

File: types/react-simple-maps/package.json (hunk 2 above)
Comment (piercemorris) [reply types/react-simple-maps/package.json:32]: Yes this is intentional, it seems the current owners are not really active, so would like to help get involved in further reviews/contributions

**Review by jakebailey — APPROVED** [review]

**Comment by piercemorris** [comment]
Ready to merge

**Pull request merged**